src/python/pants/backend/visibility/glob.py

```diff
@@ -17,6 +16,45 @@
 from pants.util.strutil import softwrap
 
 
+def is_path_glob(spec: str) -> bool:
+    """Check if `spec` should be treated as a `path` glob."""
+    return len(spec) > 0 and spec[0].isalnum() or spec[0] in "_.:/*"
```

Comment:
> Can file names begin with `-` I wonder?

I'm sure they could. Question is if we need to support picking that up as a filename when used in a text form glob? 

> Maybe it would be easier to do: `and spec[0] not in "<[("`. But, that's what the check that this function replaced had. Is there a reason to be more restrictive and not just check for these magic chars?

Reason I swapped was as I realized that adding new characters to an exclusion list is potentially breaking while for an allow list it is not, leaving more room to grow new features in the future should we need to.

Not that you asked, but the reason I broke this out to it's own function is to highlight the code implementing this rather vital piece of behaviour.

Reply:
Putting it in a separate function also makes it more readable. :+1:

Could you add a comment explaining this is about future extensibility so that future people don't try to simplify this by switching to the exclude list?